spec/GameSpec.js

```diff
@@ -0,0 +1,149 @@
+    it('appends it to the list', function () {
+      game.addFrame(frame);
+      expect(game.frameList.length).toBe(1);
+    });
+    it('calculates its score', function () {
+      game.addFrame(frame);
+      expect(game.frameScore[0]).toBe(9);
+    });
+    it('scores a previous spare', function () {
+      game.addFrame(spare);
+      game.addFrame(frame);
+      expect(game.frameScore[0]).toBe(14);
+    });
+    it('scores a previous strike', function () {
+      game.addFrame(strike);
+      game.addFrame(frame);
+      expect(game.frameScore[0]).toBe(19);
+    });
+    it('scores a previous double strike', function () {
+      game.addFrame(strike);
+      game.addFrame(strike);
+      game.addFrame(frame);
+      expect(game.frameScore[0]).toBe(24);
+    });
+  });
+
+  describe('when spare is added', function () {
+    it('appends it to the list', function () {
+      game.addFrame(spare);
+      expect(game.frameList.length).toBe(1);
+    });
+    it('delays calculating its score', function () {
+      game.addFrame(spare);
+      expect(game.frameScore[0]).toBe(null);
+    });
+    it('scores a previous spare', function () {
+      game.addFrame(spare);
+      game.addFrame(spare);
+      expect(game.frameScore[0]).toBe(16);
+    });
+    it('scores a previous strike', function () {
+      game.addFrame(strike);
+      game.addFrame(spare);
+      expect(game.frameScore[0]).toBe(20);
+    });
+    it('scores a previous double strike', function () {
+      game.addFrame(strike);
+      game.addFrame(strike);
+      game.addFrame(spare);
+      expect(game.frameScore[0]).toBe(26);
+    });
+  });
+
+  describe('when strike is added', function () {
+    it('appends it to the list', function () {
+      game.addFrame(strike);
+      expect(game.frameList.length).toBe(1);
+    });
+    it('delays calculating its score', function () {
+      game.addFrame(strike);
+      expect(game.frameScore[0]).toBe(null);
+    });
+    it('scores a previous spare', function () {
+      game.addFrame(spare);
+      game.addFrame(strike);
+      expect(game.frameScore[0]).toBe(20);
+    });
+    it('delays scoring a previous strike', function () {
+      game.addFrame(strike);
+      game.addFrame(strike);
+      expect(game.frameScore[0]).toBe(null);
+    });
+    it('scores a previous double strike', function () {
+      game.addFrame(strike);
+      game.addFrame(strike);
+      game.addFrame(strike);
+      expect(game.frameScore[0]).toBe(30);
+    });
+  });
+
+  describe('calculates the game score', function () {
+    beforeEach(function () {
+      for (var i = 0; i < 9; i++) { game.addFrame(frame) }
+    });
+    it('after frame 10 if frame 10 is normal', function () {
+      expect(game.score).toBe(null);
+      game.addFrame(frame);
+      expect(game.score).toBe(90);
+    });
+    it('after frame 11 if frame 10 is spare', function () {
+      game.addFrame(spare);
+      expect(game.score).toBe(null);
```

Comment:
Rather than constantly checking if the game score is null on frame 9, you can have 1 test at the beginning to do that